### PR TITLE
Feature: restructure platforms folder

### PR DESCRIPTION
**Pull request by lenvm**
## Detailed description

Previously the folders `stm32` and `tm4c` looked like platforms. One might therefore expect that firmware can build for the with `make PROBE_HOST=tm4c`. Actually these folders contain code common to multiple platforms.

Previously also the linker scripts of several files were stored in the `stm32` folder, which was unintuitive.

This pull request changes the following:
- It changes the folder names `stm32` to `common/stm32` and `tm4c` to `common/tm4c`.
- It moves the linker script of each platforms to the folder of that platform.
- It adds new linker scripts for the platforms that did not have their own linker script.

## Your checklist for this pull request

* [x] I've read the [Code of Conduct](https://github.com/blackmagic-debug/blackmagic/blob/main/CODE_OF_CONDUCT.md)
* [x] I've read the [guidelines for contributing](https://github.com/blackmagic-debug/blackmagic/blob/main/CONTRIBUTING.md) to this repository
* [x] It builds for hardware native (`make PROBE_HOST=native`)
* [x] It builds as BMDA (`make PROBE_HOST=hosted`)
* [x] It builds for all_platforms (`make all_platforms`)
* [x] I've tested it to the best of my ability
* [x] My commit messages provide a useful short description of what the commits do

### CODE REVIEW

**Comment by esden** [comment]
Thank you @lenvm for this. I think it is a good improvement for the codebase clarity.

I think it would be better to add another directory level here, and put the architecture common files in `common/stm32` and `common/tm4c` respectively.

**Comment by lenvm** [comment]
@esden Thanks for your feedback! I have updated the pull request accordingly.

@dragonmux can you review this pull request and merge it if it is ok?

File: src/platforms/launchpad-icdi/Makefile.inc [hunk 1]
```diff
@@ -9,11 +9,11 @@ CPU_FLAGS = -mcpu=cortex-m4 -mthumb -mfpu=fpv4-sp-d16 -mfloat-abi=hard
 CFLAGS += $(INCLUDES) $(CPU_FLAGS) -DSERIAL_NO=$(SERIAL_NO) -DTARGET_IS_BLIZZARD_RB1 -DLM4F -DPART_TM4C123GH6PM
 CFLAGS += -DDFU_SERIAL_LENGTH=9
 
-LINKER_SCRIPT="platforms/tm4c/tm4c.ld"
+LINKER_SCRIPT="platforms/common/tm4c/tm4c.ld"
```
Comment (dragonmux) [inline src/platforms/launchpad-icdi/Makefile.inc:12]: Given the other platforms have had their linker scripts moved into the platform directory, does it not make sense to do that for this one too for consistency?

**Review by dragonmux — CHANGES_REQUESTED** [review]
There's only one consistency issue we spotted, the rest of this looks great. We'll merge one that one item is addressed.

Thank you for the contribution!

File: src/platforms/launchpad-icdi/Makefile.inc (hunk 1 above)
Comment (lenvm) [reply src/platforms/launchpad-icdi/Makefile.inc]: Thanks for the feedback! I was thinking that for the `tm4c` platform there only seems to be one chip, and therefore only one ram and rom size, but out of consistency considerations I also agree it makes sense to place it in the directory of the respective platform. I have implemented this.

Additionally, for consistency, I have also renamed the linker script of the native platform.

**Review by dragonmux — APPROVED** [review]
Good call with the native platform too - happy with this, so merging.

**Review by dragonmux — APPROVED** [review]

**Comment by dragonmux** [comment]
As a note for the future, please use `git rebase -i main` rather than a merge commit. Thankfully, GitHub will do that for us in merging this.

**Pull request merged**